### PR TITLE
PRC-161 : Manage contact emergency flag

### PR DESCRIPTION
**Pull request by prabash-moj**
PRC-161 : Manage contact emergency flag

Dev test results, :

https://github.com/user-attachments/assets/e072fea7-ea76-480e-b832-5825e5f7a2b1

### CODE REVIEW

File: server/routes/contacts/manage/contact-details/contactDetailsController.test.ts [hunk 1]
```diff
@@ -268,6 +268,69 @@ describe('GET /contacts/manage/:contactId', () => {
       expect($('.relationship-active-value').text().trim()).toStrictEqual('No')
       expect($('.relationship-comments-value').text().trim()).toStrictEqual('Not provided')
     })
+
```
Comment (prabash-moj) [inline server/routes/contacts/manage/contact-details/contactDetailsController.test.ts:271]: I am hoping I can update this for other scenrios and reuse,

**Review by james-baker-moj — CHANGES_REQUESTED** [review]
Need to make work with the returnUrl stuff properly as per slack thread

File: server/views/pages/contacts/manage/contactDetails/manageRelationshipDetails.njk [hunk 2]
```diff
@@ -16,10 +16,21 @@
...
-            classes: 'emergency-contact-value'
-        }
-    },
-    {
+                 classes: 'emergency-contact-value'
+             },
+             actions: {
+                 items: [
+                     {
+                         href: prisonerContactId +"/emergency-contact",
```
[the remark this answers is not in the document]
Comment (prabash-moj) [reply server/views/pages/contacts/manage/contactDetails/manageRelationshipDetails.njk:24]: Hmm this is working with  `href: prisonerContactId +"/emergency-contact",` I noticed the prisonerContactId and added that already, 
Updated with return url, 

Was thinking something was missing :) , wanted to come back to the return url thing and completely missed that,.

File: integration_tests/e2e/manageEmergencyContact.cy.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    Page.verifyOnPage(ManageContactDetailsPage, 'First Middle Names Last').verifyShowIsEmergencyContactAs('No')
+
+    cy.verifyLastAPICall(
+      {
+        method: 'PATCH',
+        urlPath: `/contact/${contactId}/relationship/${prisonerContactId}`,
+      },
+      { isEmergencyContact: false },
+    )
+  })
```
[the remark this answers is not in the document]
Comment (prabash-moj) [reply integration_tests/e2e/manageEmergencyContact.cy.ts:56]: added that, 

I prefer more unit tests/cheerio tests than cypress tests . so skiped those back and cancel flows as I have added good cheerios tests for those. 

But also I know for back and cancel is quick and easy one , 👍

File: server/routes/contacts/manage/manageContactsRoutes.ts [hunk 4]
```diff
@@ -342,6 +343,18 @@ const ManageContactsRoutes = (
     asyncMiddleware(updateNameController.POST),
   )
 
+  const manageEmergencyContactStatusController = new ManageEmergencyContactController(contactsService)
+  router.get(
+    '/prisoner/:prisonerNumber/contacts/manage/:contactId/relationship/:prisonerContactId/emergency-contact',
+    populatePrisonerDetailsIfInCaseload(prisonerSearchService, auditService),
```
[the remark this answers is not in the document]
Comment (prabash-moj) [reply server/routes/contacts/manage/manageContactsRoutes.ts:349]: added to all the relevant places suggested in other comments as well

File: server/data/contactsApiClient.ts [hunk 5]
```diff
@@ -213,4 +214,19 @@ export default class ContactsApiClient extends RestClient {
...
     )
   }
+
+  async updateContactRelationshipById(
+    contactId: number,
+    prisonerContactId: number,
+    request: contactsApiClientTypes.UpdateRelationshipRequest,
+    user: Express.User,
+  ): Promise<contactsApiClientTypes.PatchContactResponse> {
+    return this.patch<UpdateRelationshipRequest>(
```
[the remark this answers is not in the document]
Comment (prabash-moj) [reply server/data/contactsApiClient.ts:224]: didn't notice Intellij added that :(

File: server/routes/contacts/manage/contact-details/contactDetailsController.test.ts (hunk 1 above)
Comment (prabash-moj) [reply server/routes/contacts/manage/contact-details/contactDetailsController.test.ts:271]: I will update this for relationship, next of kin , comments , active scenarios, just need to pass the right json object only, 

This test does not do much - only it is testing true or false values showing YES or NO correctly, Mostly repetitive code,

File: integration_tests/e2e/manageEmergencyContact.cy.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (james-baker-moj) [reply integration_tests/e2e/manageEmergencyContact.cy.ts:56]: yes but as we are injecting the returnUrl now the unit tests only check that it goes to what is specified. This makes sure it works in the real world

**Review by james-baker-moj — APPROVED** [review]
LGTM

**Pull request merged**